src/back/spv/writer.rs

```diff
@@ -66,7 +78,7 @@ pub struct Writer {
     annotations: Vec<Instruction>,
     writer_flags: WriterFlags,
     void_type: Option<u32>,
-    lookup_type: FastHashMap<Word, crate::Handle<crate::Type>>,
+    lookup_type_new: FastHashMap<Word, LookupType>,
```

Comment:
I believe all of the lookup tables here need to be reversed (were they copied from the front-end?). I.e. you should never need to lookup IR stuff *by* a spirv ID. You need the opposite: given IR handles and stuff, be able to produce spirv IDs that you are saving to the file. So this table (and all others) need to look like this:
```rust
lookup_type: FastHashMap<LocalType, Word>,
```

Reply:
Done this, as result, I had to modify the lib.rs to include `Hash` and `Eq` on a couple of structs.